libs/saveFiles.js

```diff
@@ -52,25 +52,32 @@ const downloadEntry = function (entry, options) {
     jar: true
   }, entry.requestOptions)
 
+  let filePromise = rq(reqOptions)
+
+  // we have to do this since the result of filePromise is not a stream and cannot be taken by
+  // cozy.files.create
+  if (options.postProcessFile) {
+    return filePromise
+      .then(data => options.postProcessFile(data))
+  }
+  return filePromise
+}
+
+const createFile = function (entry, options) {
   return cozy.files
     .statByPath(options.folderPath)
     .then(folder => {
-      let filePromise = rq(reqOptions)
       const createFileOptions = {
         name: getFileName(entry),
         dirID: folder._id,
         contentType: options.contentType
       }
 
-      // we have to do this since the result of filePromise is not a stream and cannot be taken by
-      // cozy.files.create
-      if (options.postProcessFile) {
-        return filePromise
-        .then(data => options.postProcessFile(data))
-        .then(data => cozy.files.create(data, createFileOptions))
+      if (entry.filestream) {
+        return cozy.files.create(entry.filestream, createFileOptions)
       }
 
-      return cozy.files.create(filePromise, createFileOptions)
+      return cozy.files.create(downloadEntry(entry, options), createFileOptions)
```

Comment:
I think it would be clearer if there was only one `return`.

What do you think of : 

```
const toCreate = entry.filestream || downloadEntry(entry, options)
return cozy.files.create(toCreate, createFileOptions)
```

Reply:
Personally, I prefer to return as soon as possible, but it is not my call here :)